### PR TITLE
develop: refactor PyMySQL dev guide

### PR DESCRIPTION
**Pull request by wd0517**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v7.0 (TiDB 7.0 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-sample-application-python-pymysql.md [hunk 1]
```diff
@@ -1,287 +1,287 @@
...
-        try:
-            trade_update(cursor, sell_id, buy_id, amount, price)
-        except Exception as err:
-            connection.rollback()
-            print(f'something went wrong: {err}')
-        else:
-            connection.commit()
-            print("trade success")
+    ```dotenv
+    TIDB_HOST='{host}.clusters.tidb-cloud.com'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-python-pymysql.md:134]:
```suggestion
    TIDB_HOST='{host}'  # e.g. gateway01.ap-northeast-1.prod.aws.tidbcloud.com
```

File: develop/dev-guide-sample-application-python-pymysql.md [hunk 2]
```diff
@@ -1,287 +1,287 @@
...
+    TIDB_PORT='4000'
+    TIDB_USER='{username}'
```
[the remark this answers is not in the document]
Comment (sykp241095) [reply develop/dev-guide-sample-application-python-pymysql.md:136]: and give an example @wd0517

File: develop/dev-guide-sample-application-python-pymysql.md (hunk 1 above)
Comment (sykp241095) [reply develop/dev-guide-sample-application-python-pymysql.md:134]: and give an example @wd0517

File: develop/dev-guide-insert-data.md [hunk 3]
```diff
@@ -218,7 +218,7 @@ with get_connection(autocommit=True) as connection:
 
 For complete examples in Python, see:
 
-- [Use PyMySQL to build a simple CRUD app with TiDB and Python](/develop/dev-guide-sample-application-python-pymysql.md#step-2-get-the-code)
+- [Use PyMySQL to build a simple CRUD app with TiDB and Python](/develop/dev-guide-sample-application-python-pymysql.md)
```
Comment (Oreoxmt) [inline develop/dev-guide-insert-data.md:221]:
```suggestion
- [Connect to TiDB with PyMySQL](/develop/dev-guide-sample-application-python-pymysql.md)
```

File: develop/dev-guide-sample-application-python-pymysql.md [hunk 4]
```diff
@@ -1,287 +1,287 @@
...
 
-If you need to run the script multiple times, follow the [Table initialization](#step-31-initialize-table) section to initialize the table again before each run.
+The Python driver provides low-level access to the database, but it requires the developers to:
 
-```bash
-python3 pymysql_example.py
-```
+- Manually establish and release database connections.
+- Manually manage database transactions.
+- Manually map data rows (represented as tuple/dict in `pymysql`) to data objects.
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-pymysql.md:272]:
```suggestion
- Manually map data rows (represented as a tuple or dict in `pymysql`) to data objects.
```

File: develop/dev-guide-sample-application-python-pymysql.md [hunk 5]
```diff
@@ -1,287 +1,287 @@
...
+
+Unless you need to write complex SQL statements, it is recommended to use [ORM](https://en.wikipedia.org/w/index.php?title=Object-relational_mapping) framework for development, such as [SQLAlchemy](/develop/dev-guide-sample-application-python-sqlalchemy.md), [Peewee](/develop/dev-guide-sample-application-python-peewee.md), and Django ORM. It can help you:
+
+- Reduce [boilerplate code](https://en.wikipedia.org/wiki/Boilerplate_code) for managing connections and transactions.
+- Manipulate data with data objects instead of a number of SQL statements.
+
+## Next steps
+
+- Learn more usage of `PyMySQL` from [the documentation of PyMySQL](https://pymysql.readthedocs.io).
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-pymysql.md:281]:
```suggestion
- Learn more usage of PyMySQL from [the documentation of PyMySQL](https://pymysql.readthedocs.io).
```

**Review by Oreoxmt — APPROVED** [review]

**Review by sykp241095 — APPROVED** [review]

**Pull request merged**